### PR TITLE
Created ghost object

### LINKED ISSUE #67: Interaction distance hard coded
## User story
As a developer, I want interaction distance to be linked between `PlayerInteraction` and `PlaceObject`, so that I can easily control it in one place.
## Issue
Interaction distance is a serialized field in Player Interaction script, but it is not linked with interaction distance in Place Object script which is currently hard coded.

### LINKED ISSUE #87: Create Ghost Object
## User Story
As a player, I want to know where and how an object will sit before I place it, so that I can place it how I want.

## Issue
Create a ghost object in front of the player when they are holding an item, previewing the orientation and location of that object before they place it

### PR DESCRIPTION
**Pull request by otherotherjosh**
## User Story
As a player, I want to know where and how an object will sit before I place it, so that I can place it how I want.

## Issue
Create a ghost object in front of the player when they are holding an item, previewing the orientation and location of that object before they place it (closes #87 & closes #67)

## Changes
Ghost object is a child of Main Camera on the Player prefab. When player holds an object, the ghost object inherits the held item's mesh (by combining all submeshes) and makes it use only 1 material (sets all polygons to use material 0 from mesh renderer). This is the ghost object material. It has an animator which controls the pulsing of transparency and also switches color from blue to red when player is looking at a place where the item cannot be placed.

### CODE REVIEW

File: Assets/Scripts/PlaceObject.cs [hunk 1]
```diff
@@ -3,35 +3,38 @@
...
-        if (!Physics.Raycast(transform.position, transform.TransformDirection(Vector3.forward), out hit, 10)) // hard coded interaction distance (10)
-            return;
+        if (!playerInteraction.raycastHasHit) return;
+        hit = playerInteraction.Hit;
 
         if (Vector3.Angle(hit.normal, Vector3.up) > 5f) //angle threshhold to place objects on flat surfaces only
         {
-            Debug.Log("Can't place object on non-flat surface");
+            if (placeAction.WasPressedThisFrame())
+                Debug.Log("Can't place object on non-flat surface");
```
Comment (lolyay270) [inline Assets/Scripts/PlaceObject.cs:36]: The lack of curly braces here is confusing. Maybe put it into one line?

File: Assets/Scripts/PlaceObject.cs [hunk 2]
```diff
@@ -40,10 +43,22 @@ void InstantiateObject(InputAction.CallbackContext ctx)
         {
             if (!hit.collider.CompareTag("Shelf"))
             {
-                Debug.Log("Stock items can only be placed on shelves");
+                if (placeAction.WasPressedThisFrame())
+                    Debug.Log("Stock items can only be placed on shelves");
```
Comment (lolyay270) [inline Assets/Scripts/PlaceObject.cs:47]: There are also no curly braces here

**Review by lolyay270 — APPROVED** [review]
This feature is awesome. It improves the UX exponentially.
I have noticed a couple places where the usage of curly braces in if statements is inconsistent, and I have made a comment at the exact locations.
I really liked play testing this branch. The pulsing animation gives so much life to the game

**Review by Christian-Irvine — APPROVED** [review]
Yep, looks good! Nice work, in fact. Fantastic work!

**Review by Christian-Irvine — APPROVED** [review]
Yuh, pretty good

**Review by lolyay270 — APPROVED** [review]

**Review by wilsjw — DISMISSED** [review]

**Review by lolyay270 — APPROVED** [review]

**Review by Christian-Irvine — APPROVED** [review]
Still looking good

**Review by Christian-Irvine — APPROVED** [review]
yes its great and wonderful and perfect <3

**Pull request merged**